src/filters/dmx_m2ts.c

```diff
@@ -1106,12 +1168,86 @@ static void m2tsdmx_on_event(GF_M2TS_Demuxer *ts, u32 evt_type, void *param)
 		gf_list_add(es->props, t);
 	}
 	break;
+	case GF_M2TS_EVT_ID3:
+	{
+		GF_M2TS_PES_PCK *pck = (GF_M2TS_PES_PCK*)param;
+		GF_BitStream *bs;
+		GF_M2TS_Prop *t;
+		u32 count = gf_list_count(pck->stream->program->streams);
+		for (i=0; i<count; i++) {
+			GF_M2TS_PES *es = gf_list_get(pck->stream->program->streams, i);
+			if (!es->user) {
+				GF_LOG(GF_LOG_DEBUG, GF_LOG_CONTAINER, ("[M2TSDmx] ID3 metadata not assigned to a given PID, not supported\n"));
+				continue;
+			}
+
+			// attach ID3 markers to audio
+			GF_FilterPid *opid = (GF_FilterPid *)es->user;
+			const GF_PropertyValue *p = gf_filter_pid_get_property(opid, GF_PROP_PID_STREAM_TYPE);
+			if (!p) return;
+			if (p->value.uint != GF_STREAM_AUDIO)
+				continue;
+
+			GF_SAFEALLOC(t, GF_M2TS_Prop);
+			if (!t) break;
+			t->type = M2TS_ID3;
+			bs = gf_bs_new(NULL, 0, GF_BITSTREAM_WRITE);
+			gf_bs_write_u32(bs, 90000);                     // timescale
+			gf_bs_write_u64(bs, pck->PTS);                  // pts
+			gf_bs_write_data(bs, pck->data, pck->data_len); // data
```

Comment:
Does it make sense to write `pck->data_len` before `pck->data` so that other filters that parse this data know how many bytes to read? For instance [here](https://github.com/gpac/gpac/pull/2748/files#diff-e0502e3bb4133c2be5d6f040521b18af03378ede1e3dfb4c7c71910ef2a7136cR6349)

Reply:
We know the size of the property. That being said what would happen if several properties are pushed simultaneously? The spec says that "several messages in the same 'emsg' box (i.e. same message_data[] array) shall be separated with '0x0A'." Is it a realistic use-case?